Wikipedia/Code/Panels.swift

```diff
@@ -0,0 +1,203 @@
+            })
+        }, secondaryButtonTapHandler: { sender in
+            let presenter = sender.presentingViewController
+            sender.dismiss(animated: true, completion: {
+                presenter?.wmf_showKeepSavedArticlesOnDevicePanelIfNecessary(theme: theme, completion: {
+                    WMFAuthenticationManager.sharedInstance.logout()
+                })
+            })
+        }, dismissHandler: nil)
+        present(panelVC, with: theme, animated: true, completion: nil)
+    }
+
+    @objc func wmf_showLoginOrCreateAccountToSyncSavedArticlesToReadingListPanel(theme: Theme) {
+        let panelVC = LoginOrCreateAccountToSyncSavedArticlesToReadingListPanelViewController(showCloseButton: true, primaryButtonTapHandler: { sender in
+            let presenter = sender.presentingViewController
+            sender.dismiss(animated: true, completion: {
+                presenter?.showLoginViewController(theme: theme)
+            })
+        }, secondaryButtonTapHandler: nil, dismissHandler: nil)
+        present(panelVC, with: theme, animated: true, completion: nil)
+    }
+    
+    @objc func wmf_showLoginToSyncSavedArticlesToReadingListPanelOncePerDevice(theme: Theme) {
+        guard
+            WMFAuthenticationManager.sharedInstance.loggedInUsername == nil &&
+            !UserDefaults.wmf_userDefaults().wmf_didShowLoginToSyncSavedArticlesToReadingListPanel() &&
+            !SessionSingleton.sharedInstance().dataStore.readingListsController.isSyncEnabled
+        else {
+            return
+        }
+        let panelVC = LoginToSyncSavedArticlesToReadingListPanelViewController(showCloseButton: true, primaryButtonTapHandler: { sender in
+            let presenter = sender.presentingViewController
+            sender.dismiss(animated: true, completion: {
+                presenter?.showLoginViewController(theme: theme)
+            })
+        }, secondaryButtonTapHandler: nil, dismissHandler: nil)
+        present(panelVC, with: theme, animated: true, completion: {
+            UserDefaults.wmf_userDefaults().wmf_setDidShowLoginToSyncSavedArticlesToReadingListPanel(true)
+        })
+    }
+    
+    @objc func wmf_showKeepSavedArticlesOnDevicePanelIfNecessary(theme: Theme, completion: @escaping (() -> Swift.Void) = {}) {
+        guard self.hasSavedArticles() else {
+            completion()
+            return
+        }
+        let panelVC = KeepSavedArticlesOnDevicePanelViewController(showCloseButton: false, primaryButtonTapHandler: { sender in
+            SessionSingleton.sharedInstance().dataStore.readingListsController.setSyncEnabled(false, shouldDeleteLocalLists: false, shouldDeleteRemoteLists: false)
+            sender.dismiss(animated: true, completion: nil)
+        }, secondaryButtonTapHandler: { sender in
+            SessionSingleton.sharedInstance().dataStore.readingListsController.setSyncEnabled(false, shouldDeleteLocalLists: true, shouldDeleteRemoteLists: false)
```

Comment:
@joewalsh 

This line gets called on logout when I tap "No, delete articles from device", but it doesn't seem to be working - the local lists are still there. 

<img width="431" alt="screen shot 2018-02-15 at 2 09 35 pm" src="https://user-images.githubusercontent.com/3143487/36283687-0f313ea8-125a-11e8-8234-9e1040a5c7c7.png">

Is that a bug or am I mis-reading how that method is used?

Reply:
Ah it looks like `.setSyncEnabled()` only tries to delete local lists if `enabled` is being set to `true`. Would de-coupling `shouldDeleteLocalLists` from the `enabled` parameter be ok?